exporters/otlp/testing-internal/src/main/java/io/opentelemetry/exporter/otlp/testing/internal/AbstractGrpcTelemetryExporterTest.java

```diff
@@ -562,6 +564,48 @@ void error() {
     }
   }
 
+  @Test
+  @SuppressLogger(GrpcExporter.class)
+  void errorWithStatus() {
+    addGrpcError(6, null);
+
+    TelemetryExporter<T> exporter = nonRetryingExporter();
+
+    try {
+      assertThat(
+              exporter
+                  .export(Collections.singletonList(generateFakeTelemetry()))
+                  .join(10, TimeUnit.SECONDS)
+                  .getFailureThrowable())
+          .asInstanceOf(
+              InstanceOfAssertFactories.throwable(FailedExportException.GrpcExportException.class))
+          .returns(true, Assertions.from(FailedExportException::failedWithResponse));
```

Comment:
Can we assert on the response code of the exception as well? Can we include this additional assertion in one of the existing tests like `error()` instead of introducing a new one?

Reply:
Yep no problem.